frontend/assignments/2/app.js

```diff
@@ -0,0 +1,145 @@
+const userList = require('./data').userDB;
+
+const app =  express();
+const server = http.createServer(app);
+const io = socketIo(server);
+let tweets = [];
+app.use((req, res, next) => {
+    res.setHeader('Access-Control-Allow-Origin', '*'); 
+    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, DELETE'); 
+    res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization'); 
+    if (req.method === 'OPTIONS') {
+        res.sendStatus(200); 
+    } else {
+        next(); 
+    }
+});
+
+app.use(bodyParser.json({ limit: '50mb' }));
+app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }));
+app.use(express.static(path.join(__dirname,'public')));
+
+
+app.post('/users/feed', (req, res) => {
+    console.log(req.body.username);
+    console.log(req.body.tweetContent);
+    const { username, tweetContent, imageUrl } = req.body;
+
+    const tweetId = generateTweetId();
+
+    const newTweet = {
+        id: tweetId,
+        username,
+        content: tweetContent,
+        imageUrl: imageUrl, 
+        timestamp: new Date().toISOString()
+    };
+
+    tweets.push(newTweet);
+
+    try {
+        console.log(tweets);
+        res.status(201).json({ message: 'Tweet posted successfully', tweet: newTweet });
+    } catch (error) {
+        console.error('Error sending response:', error);
+        res.status(500).json({ error: 'Internal Server Error' });
+    }
+});
+
+
+
+
+app.get('/users/feed', (req, res) => {
+    const page = parseInt(req.query.page) || 1;
+    const pageSize = parseInt(req.query.pageSize) || 5;
+    const startIndex = (page - 1) * pageSize;
+    const endIndex = startIndex + pageSize;
+
+    const paginatedTweets = tweets.slice(startIndex, endIndex);
+
+    res.json({
+        tweets: paginatedTweets,
+        totalTweets: tweets.length 
+    });
+});
+
+
+
+function generateTweetId() {
+    return Math.floor(Math.random() * 1000000) + 1;
+}
+
+
+app.get('/users/login',(req,res) => {
+    res.sendFile(path.join(__dirname,'./public/Html/login.html'));
+});
+app.get('/users/posts', (req, res) => {
+    res.sendFile(path.join(__dirname, './public/Html/view.html'));
+});
+
+app.post('/users/login', async(req,res)=>{
+    try{
+        let foundUser = userList.find((data) => req.body.username === data.user_name);
+        if(foundUser){
+
+            let submittedPassword = req.body.password;
+            let storedPassword = foundUser.password;
+            const passwordMatch = await bcrypt.compare(submittedPassword,storedPassword);
+            if(passwordMatch){
+                res.redirect(`/users/posts?username=${foundUser.user_name}`);
+            }
+            else{
+                res.send("<div align ='center'><h2>Invalid email or password</h2></div><br><br><div align ='center'><a href='./login.html'>login again</a></div>");
+            }
+        }
+        else {
+            let fakePass = `$2b$$10$ifgfgfgfgfgfgfggfgfgfggggfgfgfga`;
```

Comment:
Do not keep loose strings.
PORT number ,etc and all the integrals to be fetched from constants or process.env